phpunit/tests/fonts/font-library/wpRestFontLibraryController/installFonts.php

```diff
@@ -414,4 +414,91 @@ public function data_install_with_improper_inputs() {
 			),
 		);
 	}
+
+	/**
+	 * Tests that duplicate font faces with same font style and weight but different file extensions are merged.
+	 *
+	 * @dataProvider data_install_with_duplicate_font_faces
+	 * @param array $font_families Font families to install in theme.json format.
+	 * @param array $files         Font files to install.
+	 */
+	public function test_install_with_duplicate_font_faces( $font_families, $files = array(), $expected_response ) {
+		$install_request    = new WP_REST_Request( 'POST', '/wp/v2/fonts' );
+		$font_families_json = json_encode( $font_families );
+		$install_request->set_param( 'fontFamilies', $font_families_json );
+		$install_request->set_file_params( $files );
+
+		$response = rest_get_server()->dispatch( $install_request );
+		$this->assertSame( 200, $response->get_status(), 'The response status is not 200.' );
+		$data = $response->get_data();
+		$this->assertCount( 1, $expected_response[0]['fontFace'], 'Duplicate font faces were not removed / added incorrectly.' );
+		$this->assertStringEndsWith( $expected_response[0]['fontFace'][0]['src'], $data[0]['fontFace'][0]['src'], 'The src attribute does not match the expected font file.' );
```

Comment:
As the de-duplication functionality is not in the Rest controller but in the Font Family class, it may be better to move this test to the install tests file: `phpunit/tests/fonts/font-library/wpFontFamily/install.php`.

Apart from that, as we are not testing the API endpoint, I think the requests are not necessary because we are not testing that. Probable it would be better to add the data we want to test like this:


```
$font = new WP_Font_Family( $the_font_data_im_testing )
$font->install();
```

See this test as example: https://github.com/WordPress/gutenberg/blob/5710b3815bc4a2d29b4ab51965614a80e4e09de4/phpunit/tests/fonts/font-library/wpFontFamily/install.php#L24-L32

Reply:
Good call, done in [ce5a4a9](https://github.com/WordPress/gutenberg/pull/54490/commits/ce5a4a9cd393ebed176c6bf2df9a72be86c1daf5).